### PR TITLE
feature-rails-crud | crud operations with friendly_id gem

### PR DESCRIPTION
**Pull request by mayur-webonise**

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+class PostsController < ApplicationController
+  def index
```
Comment (kajalwebonise) [inline app/controllers/posts_controller.rb:2]: we have to use a dry concept here

add a filter method
before_action :set_post, only: %i[ show edit update destroy ]

add private method
def set_user
@post = Post.find(params[:id])
end
remove the repeating code from show edit update destroy methods

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    redirect_to root_path, status: :see_other
+  end
+
+  private
+    def post_params
+      params.require(:post).permit(:title, :content)
+    end
+
+    def mark_notifications_as_read
```
Comment (kajalwebonise) [inline app/controllers/posts_controller.rb:51]: what is the use of this method ??
from where are we getting current_user??

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Post < ApplicationRecord
+  extend FriendlyId
+friendly_id :title, use: :slugged
```
Comment (kajalwebonise) [inline app/models/post.rb:3]: check indentation

File: app/models/post.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  validates :title, presence: true
+  validates :content, presence: true, length: { minimum: 2 }
+  
+
+  def should_generate_new_friendly_id
+    title_changed? || slug.blank?
```
Comment (kajalwebonise) [inline app/models/post.rb:9]: i did not find any slug field or new migration how come it is working?

have we added any migration for friendly id?

**Comment by kajalwebonise** [comment]
@mayur-webonise  I did not find publish and unpublish a post-task ??

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (mayur-webonise) [reply app/controllers/posts_controller.rb:3]: applied DRY principle.

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit;
+  end
+
+  def update
+    if @post.update(post_params)
```
Comment (kajalwebonise) [inline app/controllers/posts_controller.rb:28]: where is post params ?

**Pull request merged**